lib/src/bazelify/arguments.dart

```diff
@@ -9,13 +9,114 @@ import 'package:args/args.dart';
 import 'package:path/path.dart' as p;
 import 'package:which/which.dart';
 
-/// Arguments when running `bazelify`, which adds Bazel support on top of pub.
+/// Shared arguments between all bazelify commands.
 class BazelifyArguments {
-  static final ArgParser _argParser = new ArgParser()
-    ..addOption(
-      'bazel',
-      help: 'A path to the "bazel" executable. Defauls to your PATH.',
-    )
+  /// The parser for shared arguments
+  static final ArgParser _argParser = () {
+    var parser = new ArgParser()
+      ..addOption(
+        'bazel',
+        help: 'A path to the "bazel" executable. Defauls to your PATH.',
+      )
+      ..addOption(
+        'package',
+        abbr: 'p',
+        help: 'A directory where "pubspec.yaml" is present. Defaults to CWD.',
+      );
+
+    parser.addCommand(
+        BazelifyInitArguments.command, BazelifyInitArguments._argParser);
+    parser.addCommand(
+        BazelifyServeArguments.command, BazelifyServeArguments._argParser);
+
+    return parser;
+  }();
+
+  static String getUsage() => _argParser.usage;
+
+  BazelifyArguments._({this.bazelExecutable, this.pubPackageDir});
+
+  /// Returns a [Future<BazelifyArguments>] impl by parsing [args].
```

Comment:
[nit] what does this tell us that the signature doesn't?

Maybe

/// Returns a concrete subtype of BazelifyArguments based on the command found in [args].

Or omit the doc?